pkgs/development/libraries/gobject-introspection/setup-hook.sh

```diff
@@ -18,14 +18,7 @@ giDiscoverSelf() {
     fi
 }
 
-# gappsWrapperArgsHook expects GI_TYPELIB_PATH variable to be set by this.
-# Until we have dependency mechanism in generic builder, we need to use this ugly hack.
-if [[ " ${preFixupPhases:-} " =~ " gappsWrapperArgsHook " ]]; then
-    preFixupPhases+=" "
-    preFixupPhases="${preFixupPhases/ gappsWrapperArgsHook / giDiscoverSelf gappsWrapperArgsHook }"
-else
-    preFixupPhases+=" giDiscoverSelf"
-fi
+postInstallHooks+=(giDiscoverSelf)
```

Comment:
I would say `preFixup` is the right place for setting variables used by `fixupPhase` so this is just changing one hack for another. Proper solution would be allowing hooks to depend on each other.

But this is probably only a cosmetic problem, as long as `giDiscoverSelf` still runs after `postInstall` passed to `mkDerivation`.